aws/cloudformation/standalone/gen_ai_curriculum/gen_ai_curriculum.yml.erb

```diff
@@ -0,0 +1,82 @@
+# Gen AI Curriculum Resources
+# This template is manually deployed to create/update Gen AI Curriculum resources.
+# TODO:
+  # Fix hardcoded execution role ARN
+
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: SageMaker resources for Gen AI Curriculum.
+
+Parameters:
+  Environment:
+    Type: String
+    Default: "test"
+    Description: "Environment being deployed to (prod or test)"
+  HuggingFaceToken:
+    Type: String
+    Description: "HuggingFace token required for deploying HuggingFace models."
+  HuggingFaceImageUri:
+    Type: String
+    Default: "763104351884.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-tgi-inference:2.3.0-tgi2.2.0-gpu-py310-cu121-ubuntu22.04-v2.0"
+    Description: "HuggingFace Image URI to use for the model."
+Resources:
+<% Config::ENDPOINT_CONFIGS.each do |config| %>
+  <%= config[:model_name] %>Model:
+    Type: AWS::SageMaker::Model
+    Properties:
+      ModelName: !Sub "<%= "#{config[:model_name]}-${Environment}-#{Time.now.to_i}" %>"
+      ExecutionRoleArn: !Sub "arn:aws:iam::${AWS::AccountId}:role/AmazonSageMaker-ExecutionRole-test"
+      PrimaryContainer:
+        Environment:
+          {
+            "HF_MODEL_ID": "<%= config[:hf_model_id] %>",
+            "HUGGING_FACE_HUB_TOKEN": !Sub "${HuggingFaceToken}",
+            "MAX_INPUT_LENGTH": 3000,
+            "MAX_TOTAL_TOKENS": 4096,
+            "SAGEMAKER_CONTAINER_LOG_LEVEL": 20,
+            "SAGEMAKER_REGION": "us-east-1",
+            "SM_NUM_GPUS": 1,
```

Comment:
Will we have to scale this in sync with our HW instance type if we end up using one with more GPUs? Not a concern, just curious.

Reply:
In theory yes - currently all of the instances we have access to only support 1 GPU, but if we want to start using larger instances we can probably parameterize this based on the instance type.